### ISSUE TITLE
Specs do not match expected in`TestFilter_NewQuery`

### ISSUE THREAD

**Issue by fchikwekwe** [influxdata/flux#2494]
Tests are failing to match expected for a couple of different reasons. 

In some cases, constraints appear to not match: 
`error compiling spec error = regexp is not of kind Equatable, wantErr false`

In other cases, the spec body does not match: 
```              		&{
              			ID: "filter1",
              			Spec: &universe.FilterOpSpec{
              				Fn: interpreter.ResolvedFunction{
              					Fn: &semantic.FunctionExpression{
              						... // 1 ignored field
              						Defaults: nil,
              						Block: &semantic.FunctionBlock{
              							... // 1 ignored field
              							Parameters: &semantic.FunctionParameters{List: []*semantic.FunctionParameter{&{Key: &semantic.Identifier{Name: "r"}}}},
            - 							Body:       &semantic.BooleanLiteral{Value: true},
            + 							Body: &semantic.Block{
            + 								loc: semantic.loc{Start: s"1:46", End: s"1:50"},
            + 								Body: []semantic.Statement{
            + 									&semantic.ReturnStatement{
            + 										loc:      semantic.loc{Start: s"1:46", End: s"1:50", Source: "true"},
            + 										Argument: &semantic.BooleanLiteral{Value: true},
            + 									},
            + 								},
            + 							},
              						},```

**Comment by fchikwekwe** [comment]
DOD: 
-  Tests are unskipped and now pass

**Comment by wolffcm** [comment]
The semantic graph in Rust can't represent a function body that is a simple expression, it needs a sequence of assignments and expression statements terminated by a `ReturnStatement`.  The fix for these is just to rewrite the function body `expr -> returnStatement{Argument: expr}`

**Comment by fchikwekwe** [comment]
UPDATE: the same issue exists for `TestMap_NewQuery`

**Comment by wolffcm** [comment]
The fix for these is very similar to #2473 and/or #2489

I'm working on those, so I'll take this one too.
